optimizely/decision_service.py

```diff
@@ -323,128 +329,216 @@ def get_variation(
         decide_reasons.append(message)
         return None, decide_reasons
 
-    def get_variation_for_rollout(self, project_config, rollout, user_id, attributes=None):
+    def get_variation_for_rollout(self, project_config, feature, user, options):
         """ Determine which experiment/variation the user is in for a given rollout.
             Returns the variation of the first experiment the user qualifies for.
 
     Args:
       project_config: Instance of ProjectConfig.
+      flagKey: Feature key.
       rollout: Rollout for which we are getting the variation.
-      user_id: ID for user.
-      attributes: Dict representing user attributes.
+      user: ID and attributes for user.
+      options: Decide options.
 
     Returns:
       Decision namedtuple consisting of experiment and variation for the user and
       array of log messages representing decision making.
     """
         decide_reasons = []
-        # Go through each experiment in order and try to get the variation for the user
-        if rollout and len(rollout.experiments) > 0:
-            for idx in range(len(rollout.experiments) - 1):
-                logging_key = str(idx + 1)
-                rollout_rule = project_config.get_experiment_from_id(rollout.experiments[idx].get('id'))
-
-                # Check if user meets audience conditions for targeting rule
-                audience_conditions = rollout_rule.get_audience_conditions_or_ids()
-                user_meets_audience_conditions, reasons_received = audience_helper.does_user_meet_audience_conditions(
-                    project_config,
-                    audience_conditions,
-                    enums.RolloutRuleAudienceEvaluationLogs,
-                    logging_key,
-                    attributes,
-                    self.logger)
+
+        if not feature:
+            return Decision(None, None, enums.DecisionSources.ROLLOUT), decide_reasons
+
+        rollout = project_config.get_rollout_from_id(feature.rolloutId)
+
+        if not rollout:
+            message = 'There is no rollout of feature {}.'.format(feature.key)
+            self.logger.debug(message)
+            decide_reasons.append(message)
+            return Decision(None, None, enums.DecisionSources.ROLLOUT), decide_reasons
+
+        rollout_rules = project_config.get_rollout_experiments(rollout)
```

Comment:
```suggestion
        rollout_rules = rollout.experiments
```

Reply:
Does not apply. We need project_config.get_rollout_experiments(rollout) to convert to array of experiment objects